pkg/install/install.go

```diff
@@ -221,7 +223,26 @@ func (i *Installer) runSteps(ctx context.Context, steps []interface{}) error {
 		switch step := step.(type) {
 		case action:
 			i.log.Printf("running step %s", runtime.FuncForPC(reflect.ValueOf(step).Pointer()).Name())
-			err = step(ctx)
+
+			timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+			defer cancel()
+
+			wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+				err = step(ctx)
```

Comment:
is it not neater to just re-run the whole install? I thought we worked to make it re-startable and idempotent for cases like this.